### PR TITLE
ENG-1612: Adds canceled state to operator lifecycle

### PR DESCRIPTION
**Pull request by vsreekanti**
## Describe your changes and why you are making these changes

This PR adds a "Canceled" state to the operator lifecycle. In particular, it does three things:
* When an upstream operator is detected to have failed, any operator that hasn't been triggered will now be marked as "Canceled".
* When an operator fails, it corresponding artifact is no longer marked as failed, it is instead marked as "Canceled" because it is never generated.
* We've also added a schema migration to update previously failed runs to reflect the new canceled state.

You can see a demo [here](https://www.loom.com/share/8aadd3e6a3c24e6a92f210ff3e38bc63).

## Related issue number (if any)

ENG-1612

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [N/A] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [x] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/engine/aq_engine.go [hunk 1]
```diff
@@ -759,8 +760,23 @@ func (eng *aqEngine) execute(
 				}
 			}
 
-			// We can continue orchestration on non-fatal errors.
+			// We can continue orchestration on non-fatal errors; currently, this only allows through succeeded operators
+			// and check operators with warning severity.
 			if shouldStopExecution(execState) {
+				log.Infof("Stopping execution of operator %v", op.ID())
```
Comment (saurav-c) [inline src/golang/lib/engine/aq_engine.go:766]: nit: lets remove these logs?

**Review by saurav-c — APPROVED** [review]
LGTM, left a few minor comments!

**Review by agiron123 — APPROVED** [review]
FE Code looks good on my end.

**Comment by vsreekanti** [comment]
Thanks for catching all my copy paste mess ups, @saurav-c!

**Pull request merged**